### PR TITLE
added naming to plots

### PR DESCRIPTION
**Pull request by berkserbet**
### Why:
* Added characteristics names to all of the plot pngs.

### What:
* Plot naming convention fixed. Old index based plotting + stitching removed.

### Checklist
- [x] Finished My Changes
- [ ] Automated Unit Tests

### CODE REVIEW

**Comment by derins** [comment]
> All this code is essentially the exact same plotting code that already exists with just a different name. It doesn't make sense to just repeat the code. Please DRY this code up.

**Comment by jongoncalves** [comment]
I still see a lot of duplicated plotting code. Am I missing something here?

**Comment by berkserbet** [comment]
> I still see a lot of duplicated plotting code. Am I missing something here?

So the amount of duplicate plotting functions is the same as the current master, Derin removed all of the older plotting functions and replaced them with the new versions. But I see what you mean, lot more simplification can me made there. I'll do some more updates.

**Comment by jongoncalves** [comment]
> > I still see a lot of duplicated plotting code. Am I missing something here?
> 
> So the amount of duplicate plotting functions is the same as the current master, Derin removed all of the older plotting functions and replaced them with the new versions. But I see what you mean, lot more simplification can me made there. I'll do some more updates.

Yeah. For example on the rbsa pipeline you see:
```
image_index = 0
        for idx, city in enumerate(normal_loadshapes.target.unique()):
            city_df = normal_loadshapes.loc[normal_loadshapes.target == city]
            max_total = city_df[['Heating', 'Cooling'] + base_enduses].sum(axis=1).max()
            max_val = 1 if max_total <= 1 else int(max_total) + 1
            for ydx, daytype in enumerate(city_df.daytype.unique()):
                title = f'{str(city)}-{str(daytype)}'
                day_df = city_df.loc[city_df.daytype == daytype]
                day_df = day_df.append(day_df.iloc[0])
                day_df = day_df.reset_index()
                day_df['Baseload'] = day_df[base_enduses].sum(axis=1)
                plot = day_df[['Baseload', 'Heating', 'Cooling']].plot(kind='area', title=title, grid=True, xticks=ticks, ylim=(0, max_val), linewidth=2, color=['black','red','blue'])
                plt.xlabel('Hour-of-Day')
                plt.ylabel('Load (pu. summer total peak)')
                fig = plot.get_figure()
                image_index_based_name = '{0:0=4d}'.format(image_index)
                fig.savefig(f'{normal_plots_dir}/{image_index_based_name}.png')
                plt.close(fig)
                image_index += 1

...

for idx, city in enumerate(normal_loadshapes.target.unique()):
            city_df = normal_loadshapes.loc[normal_loadshapes.target == city]
            max_total = city_df[['Heating', 'Cooling'] + base_enduses].sum(axis=1).max()
            max_val = 1 if max_total <= 1 else int(max_total) + 1
            for ydx, daytype in enumerate(city_df.daytype.unique()):
                title = f'RBSA-{str(city).split(",")[0]}_{str(city).split(",")[1]}-{str(daytype)}'
                day_df = city_df.loc[city_df.daytype == daytype]
                day_df = day_df.append(day_df.iloc[0])
                day_df = day_df.reset_index()
                day_df['Baseload'] = day_df[base_enduses].sum(axis=1)
                plot = day_df[['Baseload', 'Heating', 'Cooling']].plot(kind='area', title=title, grid=True, xticks=ticks, ylim=(0, max_val), linewidth=2, color=['black','red','blue'])
                plt.xlabel('Hour-of-Day')
                plt.ylabel('Load (pu. summer total peak)')
                fig = plot.get_figure()
                fig.savefig(f'{normal_plots_dir}/{title}.png')
                plt.close(fig)
```

This can all be combined

File: init.py [hunk 1]
```diff
@@ -110,32 +110,22 @@ def execute_lctk(argv):
...
 
-    image_stitcher.stitch(f'{base.LOCAL_PATH}/{mix_pipeline.run_dir}/suburban_mix/', 'suburban_mix.png')
-    image_stitcher.stitch(f'{base.LOCAL_PATH}/{mix_pipeline.run_dir}/urban_mix/', 'urban_mix.png')
-    image_stitcher.stitch(f'{base.LOCAL_PATH}/{mix_pipeline.run_dir}/mixed_mix/', 'mixed_mix.png')
-    image_stitcher.stitch(f'{base.LOCAL_PATH}/{mix_pipeline.run_dir}/rural_mix/', 'rural_mix.png')
-
-    image_stitcher.stitch(f'{base.LOCAL_PATH}/{mix_pipeline.run_dir}/suburban_mix_hour_norm/', 'suburban_mix_hour_norm.png')
-    image_stitcher.stitch(f'{base.LOCAL_PATH}/{mix_pipeline.run_dir}/urban_mix_hour_norm/', 'urban_mix_hour_norm.png')
-    image_stitcher.stitch(f'{base.LOCAL_PATH}/{mix_pipeline.run_dir}/mixed_mix_hour_norm/', 'mixed_mix_hour_norm.png')
-    image_stitcher.stitch(f'{base.LOCAL_PATH}/{mix_pipeline.run_dir}/rural_mix_hour_norm/', 'rural_mix_hour_norm.png')
```
Comment (jongoncalves) [inline init.py]: What's the reason for adding the image stitching back in?

File: pipelines/ceus/ceus.py [hunk 2]
```diff
@@ -127,7 +127,7 @@ def generate_result_plots(self):
         base_enduses.remove('daytype')
         base_enduses.remove('Heating')
         base_enduses.remove('Cooling')
-        ticks = np.arange(0, 25, 3) 
+        ticks = np.arange(0, 25, 3)
```
Comment (jongoncalves) [inline pipelines/ceus/ceus.py]: This is specified in two different places with the same value. Specify once as a class prop.

File: pipelines/ceus/ceus.py [hunk 3]
```diff
@@ -147,119 +147,76 @@ def generate_result_plots(self):
...
-                    plt.ylabel('Load (pu. summer total peak)')   
+                    plt.ylabel('Load (pu. summer total peak)')
+                    fig = plot.get_figure()
+                    fig.savefig(f'{components_plots_dir}/{title}.png')
+                    plt.close(fig)
+
+    def loadshapes_plotting(self, loadshapes, directory, base_enduses):
+        ######## Plotting helper function
+        import numpy as np
+        import matplotlib.pyplot as plt
```
Comment (jongoncalves) [inline pipelines/ceus/ceus.py]: Don't import these at the function level, put these imports up top

File: pipelines/mix/mix.py [hunk 4]
```diff
@@ -118,176 +112,53 @@ def generate_result_plots(self):
...
         logger.info('GENERATING hour_norm RURAL MIX PLOTS')
+        self.mix_type_plotting(mix_type=rural_mix_hour_norm, mix_type_name='RURAL', normalization='hour_norm', directory=rural_mix_hour_norm_plots_dir)
 
-        image_index = 0
-        for idx, city in enumerate(rural_mix_hour_norm.target.unique()):
-            city_df = rural_mix_hour_norm.loc[rural_mix_hour_norm.target == city]
+    def mix_type_plotting(self, mix_type, mix_type_name, normalization, directory):
+        ######## Plotting helper function
+        import numpy as np
+        import matplotlib.pyplot as plt
```
Comment (jongoncalves) [inline pipelines/mix/mix.py]: Same as the other comment, move these to the top of the file

File: pipelines/mix/mix.py [hunk 5]
```diff
@@ -118,176 +112,53 @@ def generate_result_plots(self):
...
             max_total = city_df[plotting_components].sum(axis=1).max()
             max_val = 1 if max_total <= 1 else int(max_total) + 1
             for ydx, daytype in enumerate(city_df.daytype.unique()):
-                title = f'{str(city)}-{str(daytype)}'
+                title = f'{mix_type_name}-{normalization}-{str(city).split(",")[0]}_{str(city).split(",")[1]}-{str(daytype)}'
                 day_df = city_df.loc[city_df.daytype == daytype]
                 day_df = day_df.append(day_df.iloc[0])
                 day_df = day_df.reset_index()
-                plot = day_df[plotting_components].plot(kind='area', title=title, grid=True, xticks=ticks, ylim=(0, 1), linewidth=2, color=['green','yellow','brown','blue','grey','black','red'])
+                plot = day_df[plotting_components].plot(kind='area', title=title, grid=True, xticks=ticks, ylim=(0, 1.2), linewidth=2, color=['green','yellow','brown','blue','grey','black','red'])
```
Comment (jongoncalves) [inline pipelines/mix/mix.py]: Spaces after the commas in the colors array.

File: pipelines/rbsa/rbsa.py [hunk 6]
```diff
@@ -145,7 +145,7 @@ def generate_result_plots(self):
         base_enduses.remove('daytype')
         base_enduses.remove('Heating')
         base_enduses.remove('Cooling')
-        ticks = np.arange(0, 25, 3) 
+        ticks = np.arange(0, 25, 3)
```
Comment (jongoncalves) [inline pipelines/rbsa/rbsa.py]: Same as above...

File: pipelines/rbsa/rbsa.py [hunk 7]
```diff
@@ -165,113 +165,73 @@ def generate_result_plots(self):
...
-                image_index_based_name = '{0:0=4d}'.format(image_index)
-                fig.savefig(f'{components_plots_dir}/{image_index_based_name}.png')
+                fig.savefig(f'{components_plots_dir}/{title}.png')
                 plt.close(fig)
-                image_index += 1   
 
+    def loadshapes_plotting(self, loadshapes, directory, base_enduses):
+        ######## Plotting helper function
+        import numpy as np
+        import matplotlib.pyplot as plt
```
Comment (jongoncalves) [inline pipelines/rbsa/rbsa.py]: Same as above

File: pipelines/rbsa/rbsa.py [hunk 8]
```diff
@@ -165,113 +165,73 @@ def generate_result_plots(self):
...
+
+        ticks = np.arange(0, 25, 3)
```
Comment (jongoncalves) [inline pipelines/rbsa/rbsa.py]: Idem

File: init.py (hunk 1 above)
Comment (berkserbet) [reply init.py]: Was testing something and must have left it in, deleted now

File: pipelines/ceus/ceus.py (hunk 2 above)
Comment (berkserbet) [reply pipelines/ceus/ceus.py]: fixed

File: pipelines/ceus/ceus.py (hunk 3 above)
Comment (berkserbet) [reply pipelines/ceus/ceus.py]: fixed

File: pipelines/mix/mix.py (hunk 4 above)
Comment (berkserbet) [reply pipelines/mix/mix.py]: fixed

File: pipelines/mix/mix.py (hunk 5 above)
Comment (berkserbet) [reply pipelines/mix/mix.py]: fixed

File: pipelines/rbsa/rbsa.py (hunk 6 above)
Comment (berkserbet) [reply pipelines/rbsa/rbsa.py]: fixed

File: pipelines/rbsa/rbsa.py (hunk 7 above)
Comment (berkserbet) [reply pipelines/rbsa/rbsa.py]: fixed

File: pipelines/rbsa/rbsa.py (hunk 8 above)
Comment (berkserbet) [reply pipelines/rbsa/rbsa.py]: fixed

**Pull request merged**